test/e2e/common/client.go

```diff
@@ -121,7 +121,6 @@ L:
 			break L
 		case <-readyChan:
 			break L
-		default:
```

Comment:
Removing this `default` changes the behavior. I think the current code doesn't wait 10 seconds because of this `default`.

Reply:
Removing the default does change the behavior, but the existing behavior is arguably incorrect. The first case statement can never match, as `default` will always be ready before 10 seconds have passed. Also, since the presence of the default branch causes the loop to spin, you will keep creating new timers in `time.After` rapidly for as long as readyChan isn't ready, so this is arguably a memory leak, too.

Of course, once the `default` branch is removed, the loop will only ever execute exactly once, as both cases will break the loop.

Reply:
I thnk we can just put the select without the loop.